forms/react-hook-form.md

```diff
@@ -0,0 +1,307 @@
+
+- **Uncontrolled Components**: React Hook Form relies on uncontrolled components. Avoid using `defaultValue` with input fields as it may lead to unexpected behaviors.
+
+- **Use Registered Fields**: Always use the register function to register your form inputs with React Hook Form. This is how the library keeps track of form state and validation.
+
+- **Validation**: Use the built-in validation provided by React Hook Form. It supports various validation rules like required, minLength, pattern, etc. You can also use custom validation functions.
+
+- **Conditional Validation**: Implement conditional validation based on the form input's value using watch and trigger functions.
+
+- **Performance Optimization**: Use the `useForm` hooks options parameter to enable shallow form state tracking (shouldUnregister: true) for better performance with large forms.
+
+- **Field Arrays**: For dynamic field arrays (e.g., repeating form fields), use the `useFieldArray` hook provided by React Hook Form.
+
+- **Error Handling**: Handle errors gracefully. Show validation errors near the input fields and provide clear error messages.
+
+- **Use Controller (Optional)**: For integrating with third-party components that require a value prop (controlled components), use the `Controller` component from React Hook Form.
+
+- **Form Submission**: Use the `handleSubmit` function from the useForm hook to handle form submissions. Avoid using the native form submit unless necessary.
+
+- **Disable Browser Validation**: To avoid conflicts with the built-in browser validation, remember to set the noValidate attribute on your form element.
+
+- **Clear Errors**: Clear errors when necessary, especially if you're using conditional fields or dynamically showing/hiding inputs.
+
+- **Test Your Forms**: Write unit tests to ensure your forms behave as expected, especially concerning validation and form submission.
+
+- **Keep Updated**: Stay updated with the React Hook Form documentation and the library updates to leverage new features and improvements.
+
+## Configure form fields
+
+```ts
+import { FC } from 'react';
+import { useForm } from 'react-hook-form';
+
+const  MyFormComponent: FC = () => {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+
+  const onSubmit = (data) => {
+    console.log(data); // Data contains the form input values
+  };
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      {/* Form fields go here */}
+      <label htmlFor="name">Name:</label>
+      <input
+        type="text"
+        id="name"
+        {...register('name', { required: 'Name is required' })}
+      />
+      {errors.name && <p>{errors.name.message}</p>}
+
+      {/* Add more form fields here */}
+
+      <button type="submit">Submit</button>
+    </form>
+  );
+}
+
+export default MyFormComponent;
+```
```

Comment:
This example of code is similar to the previous one. Maybe we need to combine them or use just one example